DigitalLearningSolutions.Data.Tests/Services/CourseServiceTests.cs

```diff
@@ -183,7 +183,7 @@ public void
             {
                 CentreId = 2,
                 CourseCategoryId = 2,
-                AllCentres = true,
+                AllCentres = false,

```

Comment:
Oh, sorry, I get it now. We are changing the logic. The new logic is only equivalent to the original if we assume that customisations will either be `allCentres` or will have a `centreId` matching the user's (and not both). But actually, the centre that made the customisation should be able to manage it, even if it is `allCentres`.

So I need to check something with Kevin (I'll do that on Monday), but I _think_ the fix will be to change the `all centres course => false` line to `all centres course && course.centreId != user.centreId => false`.

The case we're concerned about is where the user's centre matches the centre on an All Centres customisation. Presumably the user should be able to manage that course. I'll get back to you on Monday.

Reply:
Confirmation from Kevin around `AllCentres` courses:
* admins at the centre specified by Customisation.CentreId should be able to view and manage the course.
* admins at other centres should be able to view but **not** manage the course.

Reply:
Almost there I think. Just one case that concerns me.
Can you write a test that AdminUserCanViewCourse returns true if all of the following are satisfied?
* The course is at their centre
* The course matches their categoryId
* The course is an all-centres course
* courseValidationDetails.CentreHasApplication = false.

At the moment I think it will fail, but it should pass.